packages/testcontainers/src/generic-container/generic-container.test.ts

```diff
@@ -123,6 +123,15 @@ describe("GenericContainer", () => {
     expect(output).toEqual(expect.stringContaining("/tmp"));
   });
 
+  it("should set platform", async () => {
+    const container = await new GenericContainer("amd64/alpine")
```

Comment:
I try to do something like the code below to get the architecture of the image only that keeps failing and I don't understand. Some help would be appreciated.

```typescript
 const { output } = await container.exec(["uname", "-a"]);
expect(output).toEqual(expect.stringContaining("x86_64"));
```

Reply:
A few things:
1. Starting a container and running an exec command isn't equivalent to `docker run`. You would need to keep the container running. Something like:
    ```js
    .withEntrypoint(["bash"])
    .withCommand(["-c", "echo 'START'; sleep infinity"])
    ```
2. The image `amd64/alpine` doesn't expose ports `8080`, so you'll want to remove `withExposedPorts(8080)` or else Testcontainers will assume the container has failed to start as it never exposes this port
3. This test won't work on ARM machines so I'm not sure it's possible to have a test for it